core/components/ui/date-picker/date-picker.tsx

```diff
@@ -41,7 +41,7 @@ export const DatePicker = React.forwardRef<React.ElementRef<'div'>, DatePickerPr
           <PopoverTrigger asChild>
             <Input
               placeholder={placeholder}
-              readOnly={true}
+              readOnly={!required}
```

Comment:
**Note**:  If the element is `read-only`, then the element's value can not be updated by the user, and does not participate in constraint validation. So I've made this change to validate against `missingValue` when the field is required.